### PR TITLE
docs: Add interactive example for `Anchor`

### LINKED ISSUE #2510: Interactive example that explain how `Anchor` works
# What could be improved

Since many people are confused about how the `Anchor` works and since we have some great tools to visually explain things in the docs now I think it would be great to have an interactive example to explain it.

I'm imagining an example with buttons that change the anchor and you can visually see where the position is etc.
Possibly two examples, one nested with a child and one without.

# Why should this be improved

To help people understand the `Anchor` better.

# Any risks?

No.

# More information

The `RouterComponent` example is pretty good:
https://docs.flame-engine.org/latest/flame/router.html

### PR DESCRIPTION
**Pull request by ufrshubham**
# Description

This PR add an interactive example to show the parent child relationship when their anchors are changed. It also elaborates the same in written docs as well.

https://user-images.githubusercontent.com/33748002/235361268-7636d25c-5c07-4124-9ce5-de9485b0fb1b.mp4

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [-] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #2510

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Looks good! Maybe it could be good to clarify that the position is 0,0 for the child?

**Comment by ufrshubham** [comment]
> Looks good! Maybe it could be good to clarify that the position is 0,0 for the child?

Should I updated that in the infobox comment under the example?

**Comment by spydon** [comment]
> Should I updated that in the infobox comment under the example?

Wherever you think it looks the best :)

File: doc/flame/components.md [hunk 1]
```diff
@@ -375,12 +375,48 @@ Right/East| pi/2         | 90
 
 ### Anchor
 
+```{flutter-app}
+:sources: ../flame/examples
+:page: anchor
+:show: widget code infobox
+This example shows effect of changing `anchor` point of parent (red) and child (blue)
+components. Tap on them to cycle through the anchor points. Note that the local
+position of child component is (0, 0) at all times.
```
Comment (spydon) [inline doc/flame/components.md:384]:
```suggestion
position of the child component is (0, 0) at all times.
```

**Review by spydon — APPROVED** [review]

**Pull request merged**